lib/Dialect/Arc/Transforms/Dedup.cpp

```diff
@@ -728,8 +725,13 @@ void DedupPass::replaceArcWith(DefineOp oldArc, DefineOp newArc) {
   auto newArcName = SymbolRefAttr::get(newArc.getSymNameAttr());
   for (auto callOp : oldUses) {
     callOp.setCalleeFromCallable(newArcName);
-    newUses.push_back(callOp);
+    newUses.insert(callOp);
   }
+
+  oldArc.walk([&](mlir::CallOpInterface callOp) {
+    if (auto defOp = dyn_cast_or_null<DefineOp>(callOp.resolveCallable()))
```

Comment:
Aha, I'll look into it tomorrow!

Reply:
Thanks for the incredibly fast feedback! 🙌

Reply:
resolveCallable without a symbol table also requires the pass to be a builtin.module pass rather than a narrower pass.  I didn't check if it is already or not.

Reply:
Thanks for the information, and for having a look, @darthscsi!

I don't have the insight to tell whether promoting to class member or passing the `SymbolTable` along from `runOnOperation` is better, so I made an attempt at the latter.

Reply:
Looks good! Thanks again!

Reply:
Thanks for guiding me through!